voms-admin-server/resources/scripts/configure/voms_mysql_util.py

```diff
@@ -18,235 +18,265 @@
-            return False
-        else:
-            error_and_exit("Error checking schema existence: %s" % err_msg)
+    match = re.match("ERROR 1049", err_msg.strip())
+    if match:
+        return False
+    error_and_exit(f"Error checking schema existence: {err_msg}")
 
 
 def create_db(options):
-    print "Creating database %s" % options.dbname
-
+    print(f"Creating database {options.dbname}")
     if db_exists(options):
-        print "Schema for database %s already exists, will not create it..." % options.dbname
+        print(f"Schema for database {options.dbname} already exists, "
+              "will not create it...")
     else:
         mysql_cmd = build_mysql_command_preamble(options)
         # The database is not there, let's create it
-        mysql_proc = subprocess.Popen(
-            mysql_cmd, shell=True, stdin=subprocess.PIPE)
-        print >>mysql_proc.stdin, "create database %s;" % options.dbname
-        mysql_proc.stdin.close()
-        status = mysql_proc.wait()
+        with subprocess.Popen(mysql_cmd, shell=True,
+                              stdin=subprocess.PIPE) as mysql_proc:
+            print("create database {options.dbname};", file=mysql_proc.stdin)
+            status = mysql_proc.wait()
         if status != 0:
-            error_and_exit("Error creating MySQL database %s: %s" %
-                           (options.dbname, mysql_proc.stdout.read()))
+            error_and_exit("Error creating MySQL database {options.dbname}: "
+                           "mysql_proc.stdout.read()")
 
     grant_rw_access(options)
-
-    print "Done."
+    print("Done.")
 
 
 def drop_db(options):
-    print "Dropping database %s" % options.dbname
+    print(f"Dropping database {options.dbname}")
     if not db_exists(options):
-        print "Schema for database %s does not exist, exiting..." % options.dbname
-        exit(1)
+        print(f"Schema for database {options.dbname} does not exist, "
+              "exiting...")
+        sys.exit(1)
     else:
         mysql_cmd = build_mysql_command_preamble(options)
-        mysql_proc = subprocess.Popen(
-            mysql_cmd, shell=True, stdin=subprocess.PIPE)
-        print >>mysql_proc.stdin, "drop database %s;" % options.dbname
-        mysql_proc.stdin.close()
-        status = mysql_proc.wait()
-        if status != 0:
-            error_and_exit("Error dropping MySQL database %s: %s" %
-                           (options.dbname, mysql_proc.stdout.read()))
-    print "Done."
+        with subprocess.Popen(
+                mysql_cmd, shell=True, stdin=subprocess.PIPE) as mysql_proc:
+            print(f"drop database {options.dbname};", file=mysql_proc.stdin)
+            status = mysql_proc.wait()
+            if status != 0:
+                output = mysql_proc.stdout.read()
+                msg = f"Error dropping MySQL database {options.dbname}: {output}"
+                error_and_exit(msg)
+    print("Done.")
 
 
 def grant_rw_access(options):
```

Comment:
I don't know why git is not highlighting this line. In the main branch, the `options` argument is not defined and thus I cannot understand how this could work https://github.com/italiangrid/voms-admin-server/blob/5602fabbf004fbdc2a620613fa938fb32de39f75/voms-admin-server/resources/scripts/configure/voms_mysql_util.py#L190

I added the argument because how I guess that function was supposed to be used.